README.md

```diff
@@ -1 +1,232 @@
+*onLoaded*: It will put `${action.target}Loading` in false. Also supports selectors.  
+
+_Spoiler alert_: The last ones are useful to implement SUCCESS/FAILURE async actions pattern in thunk actions. Soon will be available this creator.  
+
+We are currently writting some other effects:  
+```
+onSetValue
+onReset
+onDelete
+onToggle
+onAppend
+onCycle
+onMap
+```
+
+And other utilities like  
+```
+composeEffects
+```
+
+New effects are welcome ! Feel free to open an issue or even a PR.  
+
+## Creators [WIP]  
+There are a few creators that also ease writing Redux reducers and actions.  
+
+*createReducer*: Receives an initialState and a reducer description. It replaces the traditional switch.  
+  Example:  
+  ```
+  const reducerDescription = {
+    'ACTION_NAME': (state, action) => ({ ...state, aTarget: ':)' })
+  }
+
+  const initialState = { aTarget: null };
+  const reducer = createReducer(initialState, reducerDescription);
+  ...
+  dispatch({ type: 'ACTION_NAME' }); // state.aTarget = ':)'
+  ```
+*createAction*: Receives a type, a target and optionally a payload. Returns a new action.  
+*createTypes*: Receives a string list and another string to prepend a namespace.  
+  Example:  
+  ```
+  const actions = createTypes(['ACTION1', 'ACTION2'], '@@NAMESPACE');
+  actions.ACTION1 === '@@NAMESPACE/ACTION1';
+  ```
+
+We are currently working on these creators:  
+```
+createThunkAction
+createMultiTargetAction
+createPollingAction
+```
+Since state handling is decoupled from its state, we could create some more complex async actions, or even map
+an effect with an action type to create families of actions.  
+More crazy and useful ideas are welcome too !  
+
+## Completers [WIP]
+You could use completers to reduce your code size.  
+
+*completeState*: Receives a state description and a list of target exceptions.  
+  Example:  
+  ```
+  const initialLongState = {
+    thing: null,
+    thingLoading: false,
+    thingError: null,
+    otherThing: null,
+    otherThingLoading: false,
+    otherThingError: null,
+    anotherThing: null
+  };
+
+  const initialStateDescription = {
+    thing: null,
+    otherThing: null,
+    anotherThing: null
+  }
+
+  const initialState = completeState(initialStateDescription, ['anotherThing']);
+
+  initialState and initialLongState are equivalent.
+  ```
+*completeReducer*: Receives an object with `primaryActions` that is a string list of action names, and optionally a `override`.
+For those actions in `primaryActions`, it will add `onLoading`, `onSuccess` and `onFailure` for `action.type`, `${action.type}_SUCCESS` and `${action.type}_FAILURE` respectively.  
+  Example:  
+  ```
+  const actions = createActions(['FETCH', 'FETCH_SUCCESS', 'FETCH_FAILURE', 'OTHER'], '@@API');
+  const reducerDescription = {
+    primaryActions: [actions.FETCH],
+    override: {
+      [actions.OTHER]: () => console.log('Completing the reducer description')
```

Comment:
I know that this is an example, but should the fuction return an object with the state? The example may be confusing